### PR TITLE
eNikshay custom setup work

### PR DESCRIPTION
**Pull request by esoergel**
Basically this adds some custom validation and modification to the user and location creation and update workflows.  Detailed spec [here](https://docs.google.com/document/d/1Pr19kp5cQz9412Q1lbVgszeZJTv0XRzizb0bFHDxvoA/edit#heading=h.u2p3r8ahp88z).

This isn't quite done yet, but it's all feature flagged, so I figure it makes sense to get this merged in and finish the rest in smaller chunks.
@nickpell (I know this is kinda big, so lemme know if you wanna walk through it in person)

### CODE REVIEW

File: custom/enikshay/__init__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from corehq.preindex import ExtraPreindexPlugin
```
Comment (dimagimon) [inline custom/enikshay/__init__.py]: F401 'ExtraPreindexPlugin' imported but unused

File: custom/enikshay/__init__.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+from django.apps import AppConfig
+from django.conf import settings
```
Comment (dimagimon) [inline custom/enikshay/__init__.py]: F401 'settings' imported but unused

File: custom/enikshay/user_setup.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
+# TODO Is this going to be translated, or English only?
```
Comment (esoergel) [inline custom/enikshay/user_setup.py]: This file has several TODOs that are intentional.

**Review by nickpell — APPROVED** [review]
Looks good to me, but tests are timing out.  Maybe merge master once another travis worker is added?

**Pull request merged**